executors/rust/Cargo.toml

```diff
@@ -12,15 +12,29 @@ env_logger = "0.9.1"
 
 substring = "1.0"
 json = "0.12.4"
-serde = { version = "1.0.143", features = ["derive"] }
+serde = {version = "1.0.143", features = ["derive", "alloc"], optional=true}
 serde_json = "1.0.48"
 rustc_version_runtime = "0.1.*"
     
-fixed_decimal = {version="0.5.0"}
-writeable =  {version="0.5.0"}
-
-icu = { version="1.0.0", features = ["serde"] }
-icu_testdata = { version="1.0.0", features = ["metadata"] }
-icu_provider = { version="1.0.0"}
-
+fixed_decimal = {version="0.5.3"}
+writeable = {version="0.5.2"}
+icu_compactdecimal = {version="0.2.0"}
+icu_displaynames = {version="0.10.0"}
+    
+icu = {version="1.2.0", features = ["serde"] }
+icu_testdata = {version="1.2.0"}
+icu_compactdecimal_data = {git="https://github.com/unicode-org/icu4x", rev="5fcb6c19f72b2e5ad1d49b1a2332774c6d68c25d", optional=true}
+icu_provider = {version="1.2.0", features=["macros"]}
+icu_locid = {version="1.2.0"}
+icu_locid_transform = {version = "1.2.0", optional=true}
+icu_plurals = { version = "1.2.0", optional=true}
+icu_decimal = { version = "1.2.0", optional=true}
+databake = { version = "0.1.3", features = ["derive"], optional = true}
+zerovec = { version = "0.9.4", features = ["yoke"] }
+    
 [features]
+default = ["compiled_data"]
+std = ["fixed_decimal/std", "icu_decimal/std", "icu_plurals/std", "icu_provider/std"]
+serde = ["dep:serde", "zerovec/serde", "icu_decimal/serde", "icu_plurals/serde"]
+datagen = ["std", "serde", "dep:databake", "zerovec/databake"]
+compiled_data = ["dep:icu_compactdecimal_data", "dep:icu_locid_transform", "icu_decimal/compiled_data", "icu_plurals/compiled_data"]
```

Comment:
Issue: As a client crate you shouldn't need features. It looks like you copied these from an ICU4X crate and they don't make sense here.